app/models/user.rb

```diff
@@ -138,7 +138,7 @@ def mhv_account_type
   end
 
   def mhv_correlation_id
-    identity.mhv_correlation_id || mpi.mhv_correlation_id
+    identity.mhv_correlation_id || mpi_profile&.mhv_correlation_id
```

Comment:
If you have the delegate definition for this, why not use it here? `mpi_mhv_correlation_id`